cdap-security/src/main/java/co/cask/cdap/security/server/ExternalAuthenticationServer.java

```diff
@@ -110,11 +110,21 @@ public ExternalAuthenticationServer(CConfiguration configuration, SConfiguration
    * @return InetSocketAddress of server.
    */
   public InetSocketAddress getSocketAddress() {
-    return new InetSocketAddress(address, port);
+    if (!server.isRunning()) {
+      return new InetSocketAddress(address, port);
+    } else {
+      // assumes we only have one connector
+      final Connector connector = server.getConnectors()[0];
+      return new InetSocketAddress(connector.getHost(), connector.getLocalPort());
+    }
   }
 
   @Override
   protected void run() throws Exception {
+    server.start();
+
+    // assumes we only have one connector
+    final Connector connector = server.getConnectors()[0];
```

Comment:
Yup, we set them up when creating the server: https://github.com/caskdata/cdap/pull/1841/files#diff-875ec12d39f8e6b2cf4725b9ec4b4305R193 and https://github.com/caskdata/cdap/pull/1841/files#diff-875ec12d39f8e6b2cf4725b9ec4b4305R198.